attachments/views.py

```diff
@@ -61,13 +72,29 @@ def attach(request, session_id):
                     session.data = {key: value}
             session.save()
             return JsonResponse({'ok': True, 'file_name': f.name, 'file_size': f.size}, content_type=content_type)
+        except FileSizeException, ex:
+            return JsonResponse({'ok': False, 'error': unicode(ex)}, content_type=content_type)
+        except FileTypeException, ex:
+            return JsonResponse({'ok': False, 'error': unicode(ex)}, content_type=content_type)
         except VirusFoundException as ex:
-            logger.exception(str(ex))
-            return JsonResponse({'ok': False, 'error': force_text(ex)}, content_type=content_type)
+            user = getattr(request, 'user', None)
+            log_message = loader.render_to_string('attachments/virus_email.txt', 
+                                                  {'user': user,
+                                                   'filename' : f.name,
+                                                   'virus_signature' : virus[path][1],
+                                                   })
+            logger.exception(log_message)
+            #if ATTACHMENTS_VIRUS_EMAIL is set to a list/tuple of email addresses to send to it will send email alert
+            if getattr(settings, 'ATTACHMENTS_VIRUS_EMAIL', False):
+                #send email to email list
+                email_list = getattr(settings, 'ATTACHMENTS_VIRUS_EMAIL')
+                subject = loader.render_to_string('attachments/virus_subject.txt', {'user': user })
+                send_mail(subject, log_message, settings.DEFAULT_FROM_EMAIL, email_list)
         except Exception as ex:
             logger.exception('Error attaching file to session %s', session_id)
             return JsonResponse({'ok': False, 'error': force_text(ex)}, content_type=content_type)
     else:
+        file_uploaded.send(sender=f, request=request, session=session)
```

Comment:
Is there a reason this was moved? I don't know much about this signal, but it's probably worth checking with Noah/Jake/etc. to see if they know if/how it's being used. My concern would be that the signal is being used somewhere to manipulate the session or file object somehow.

Reply:
Chances are this will be moving back, it doesn't really make sense where it got moved (into the GET portion of the view). Checking with Lewis